src/cli/self_update/unix.rs

```diff
@@ -174,15 +174,15 @@ fn remove_legacy_paths(process: &Process) -> Result<()> {
     remove_legacy_source_command(
         format!(
             "export PATH=\"{}/bin:$PATH\"\n",
-            shell::cargo_home_str(process)?
+            Posix.cargo_home_str(process)?
```

Comment:
Should we pass some kind of shell into here? Why is it okay to hardcode the shell here?

Reply:
In legacy error notes we *assume* Posix shell. You can click to expand the context to get a better view.

Reply:
Fair enough!